crates/nu-command/src/bytes/starts_with.rs

```diff
@@ -60,13 +61,73 @@ impl Command for BytesStartsWith {
             pattern,
             cell_paths,
         };
-        operate(
-            starts_with,
-            arg,
-            input,
-            call.head,
-            engine_state.ctrlc.clone(),
-        )
+
+        match input {
+            PipelineData::ExternalStream { stdout, span, .. } if stdout.is_some() => {
+                let stream = stdout.expect("stdout is some");
+
+                let mut i = 0;
+
+                for item in stream {
+                    match item {
+                        Ok(v) => match v {
+                            Value::String { val, span } => {
```

Comment:
I found this double-nested match a little hard to follow, and it would be nice to eliminate the duplicated code in the `Value::String` and `Value::Binary` branches. What about something like this instead?

```rust
let bytes: Vec<u8> = match item {
    Ok(Value::String { val, span }) => val.as_bytes().to_vec(),
    Ok(Value::Binary { val, span }) => val,
    Ok(Value::Error { error }) => todo!(), // early return
    Ok(other) => todo!(), // early return
    Err(err) => return Err(err), // early return
};
// check whether `bytes` matches` the remaining bytes in `pattern`
````

Reply:
Implemented this suggestion and it does reduce code duplication in my branches. It's better, thanks!

I chose not to collect into a Vec<u8> in the string case, and instead matched on &item so the byte slice would live as long as `item`. This means that the cloning is required on the error case, though, which sounds like a reasonable tradeoff.